### PR TITLE
Import shape function, fix deprecation warnings

### PR DESCRIPTION
**Pull request by ranjanan**
Base has a `shape` function, so there would be a conflict. This fixes that.

### CODE REVIEW

**Comment by ranjanan** [comment]
@simonbyrne any idea why coverage has decreased by 42% ( 😱 )?

**Comment by jmxpearson** [comment]
Tests are currently failing on nightly because of timeout on Travis. It's on my to-do list, but any help would be appreciated. As far as I can tell, it's mostly due to deprecation warnings external to Distributions.

**Comment by ranjanan** [comment]
@jmxpearson Which packages do you think the external deprecations are coming from? I'll try pushing to fix them.

**Comment by jmxpearson** [comment]
Actually, looking at the Travis log, there appear to be a few instances of `slice` remaining (should be `view`). Is Compat currently handling that?

Also: WARNING: Method definition (::Type{ArrayViews.UnsafeContiguousView})(Ptr{#T<:Any}, Int64, Tuple{Vararg{Int64, #N<:Any}}) in module ArrayViews at /home/travis/.julia/v0.5/ArrayViews/src/arrviews.jl:18 overwritten at /home/travis/.julia/v0.5/ArrayViews/src/arrviews.jl:26.

Other than that, things look okay. Note also that #505 is also slowing you down, since one of the workers is quitting.

**Comment by jmxpearson** [comment]
This is great. Thanks! I found another test error that is failing on my machine for nightly. Once I can get that fixed, I'll merge this, #505, and that fix together.

**Comment by ranjanan** [comment]
I'm actually sending a Compat PR (https://github.com/JuliaLang/Compat.jl/pull/238) to fix the view issue so it's compatible on release. Once that is merged, I might have to add a commit here saying `import Compat.view

**Comment by jmxpearson** [comment]
Great. I'll hold off. I'll probably also ask you to squash before I merge.

**Comment by ranjanan** [comment]
@jmxpearson since the new Compat version has now been tagged and merged into METADATA, could you restart Travis please?

File: src/Distributions.jl [hunk 1]
```diff
@@ -7,6 +7,7 @@ using StatsFuns
 using StatsBase
 using Compat
 
+import Compat.view 
 import Base.Random
 import Base: size, eltype, length, full, convert, show, getindex, scale, scale!, rand, rand!
 if VERSION > v"0.4"
```
Comment (jmxpearson) [inline src/Distributions.jl]: Don't you want something like:

```julia
if VERSION < v"0.5.0-"
    export shape
else
    import Base.shape
end
```

**Comment by jmxpearson** [comment]
I could, but I don't think it will matter much until #507 is dealt with. My plan is to work on that this morning, then merge this and #508. After that, we should be good with 0.5.

File: src/Distributions.jl (hunk 1 above)
Comment (ranjanan) [inline src/Distributions.jl]: If it's v0.4, it's going to pick from compat. Nothing happens in v0.5. So this would work.

Comment (andreasnoack) [inline src/Distributions.jl]: I guess it would be a problem that
```jl
julia> v"0.4.1"> v"0.4"
true
```

Comment (ranjanan) [inline src/Distributions.jl]: @andreasnoack you're right. Let me change that to ` < v"0.5.0-" ` instead.

**Comment by jmxpearson** [comment]
Restarted Travis, since JuliaLang/julia#17185 merged. If all goes well, will close #507.

**Comment by jmxpearson** [comment]
Looks like I jumped the gun. I may need to give nightly [a little time](https://status.julialang.org/) to percolate.

**Comment by jmxpearson** [comment]
@ranjanan Master is throwing a real error now. Cf the Travis logs, but I think it's complaining about `shape(d::Erlang)`.

**Comment by ranjanan** [comment]
The error log says `indicesbehavior(d::Erlang)`

**Comment by jmxpearson** [comment]
Right, but the trace seems to relate it to shape. I don't get this error on the jp/partests branch, where I've made most of the same fixes. If you don't have time to look at it right now, I can include your changes from here in that PR  (#508).

**Comment by ranjanan** [comment]
@jmxpearson I apologize, I'm actually travelling a fair bit right now, so I might not be as responsive as I want to be. Please feel free to take commits/whatever changes you need. Thanks!

**Comment by jmxpearson** [comment]
No worries. I'll take a look.

**Comment by jmxpearson** [comment]
Superseded by #508

**Pull request closed without merging**